### PR TITLE
Allow user to specify filename for download (#5)

### PR DESCRIPTION
**Pull request by colebillys19**
Added functionality for "Allow user to specify filename for download" (Issue #5)

I started by creating a "GenerateGifForm" component. This form contains an input for specifying the file name for download, an input for specifying GIF title text (other members of our group are currently working on this and will use this same component), and the "Generate GIF" button that previously lived in the "Preview" component. This button now acts as a submit button for the form.

Next, I changed the render method on the Preview component so that the form is only shown when both the following conditions are met:
- there are frames present in state (from which to generate a GIF)
- there is no GIF data in state (the form should not show when a GIF has already been generated) 
Additionally, I added functionality so that after a user generates a GIF, a success message is rendered.

The file name lives in the `images` state slice (as `gifsmos.gif` initially). I added an action-type (UPDATE_GIF_FILENAME), action (updateGifFileName), and reducer case.

Chris, don't look at this until Alissa approves!
@ctlusto @alissarenz

### CODE REVIEW

**Comment by colebillys19** [comment]
Also wanted to mention that the form has access to the updateGifFileName action via a prop passed from the preview component. Curious if the form should have access to the action directly via mapStateToProps (would need a container). Or is it fine as is?

File: src/components/GenerateGifForm.js
Comment (mmmaaatttttt) [inline src/components/GenerateGifForm.js]: @colebillys19 @lopezh18 @phillipson808 - since we're storing this info in redux, this component shouldn't need any local component state. (@colebillys19 you don't need to worry about this for now, Haley and Spencer are working on it)

File: src/components/GenerateGifForm.js
Comment (mmmaaatttttt) [inline src/components/GenerateGifForm.js]: i don't think this comment adds much value :) i'd nix.

File: src/components/GenerateGifForm.js
Comment (mmmaaatttttt) [inline src/components/GenerateGifForm.js]: i wouldn't worry about clearing the form on submit -- if the user adds more frames after the initial download, we can pre-populate the old form values from redux. But if they clear out the frames, we should also clear out this form state from redux.

**Comment by colebillys19** [comment]
Happy to write some testing for this feature as well.

File: src/components/GenerateGifForm.js
Comment (mmmaaatttttt) [inline src/components/GenerateGifForm.js]: here you'll need to dispatch an action rather than setting some state.

File: src/components/Sidebar.js
Comment (mmmaaatttttt) [inline src/components/Sidebar.js]: since you're destructuring props on L34, i'd just extract `gifFileName` as well, so that you can remove `this.props` on this line.

File: src/constants/action-types.js
Comment (mmmaaatttttt) [inline src/constants/action-types.js:16]: @lopezh18 @phillipson808 -- take note of @colebillys19 action type here, and make sure your name is similar (e.g. `UPDATE_GIF_TITLE` or something

File: src/reducers/images.js
Comment (mmmaaatttttt) [inline src/reducers/images.js]: Question for @ctlusto - is there a reason why you prefer spreading out a new object here, rather than just adding keys directly to the return object? e.g. a reason for returning

```js
{
  ...state,
  ...{
    gifFileName: payload.gifFileName
  }
}
```

instead of returning

```js
{
  ...state,
  gifFileName: payload.gifFileName
}
```

?

**Review by mmmaaatttttt — CHANGES_REQUESTED** [review]
@colebillys19 there's one tweak i think you can make here re: props destructuring, otherwise we should wait on approval for this until @lopezh18 and @phillipson808 push up their changes.

@ctlusto - i pinged you re: a quick question for you on style in the reducers when you get a chance.

let me know when there's more for me to review!

File: src/actions/index.js
Comment (mmmaaatttttt) [inline src/actions/index.js:66]: linebreak

Comment (mmmaaatttttt) [reply src/actions/index.js:66]: also i'd call the parameter `gifFileName` so that you can use object shorthand notation on the line with the payload.

File: src/components/GenerateGifForm.js
Comment (mmmaaatttttt) [inline src/components/GenerateGifForm.js]: remove this -- leave the text as-is unless the user clicks on the reset button (in which case you can dispatch this reset action)

File: src/components/GenerateGifForm.js
Comment (mmmaaatttttt) [inline src/components/GenerateGifForm.js]: remove this as well, for the same reason

File: src/components/GenerateGifForm.js
Comment (mmmaaatttttt) [inline src/components/GenerateGifForm.js]: This doesn't feel very DRY to me - @alissarenz, thoughts on refactoring?

File: src/components/Preview.js
Comment (mmmaaatttttt) [inline src/components/Preview.js:93]: we're passing so much stuff in here, i'm wondering whether it would be better to just connect the form. would that simplify the preview container? what do y'all think?

File: src/reducers/images.js
Comment (mmmaaatttttt) [inline src/reducers/images.js]: i'd probably use black here.

File: src/reducers/images.js
Comment (mmmaaatttttt) [inline src/reducers/images.js:82]: open question as to whether we want this pattern or whether we should spread out a new object with `fontColor`

**Review by mmmaaatttttt — CHANGES_REQUESTED** [review]
@colebillys19 @lopezh18 @phillipson808 some feedback for you in here, let me know what you think. Also i think you should be able to mark old feedback as resolved once you resolve it. 

I'll leave this open until issues are resolved, and you've added testing and the title text preview. let me know what else you need from me! Once this is good to pass up the chain we should also squash some of these commits.

File: src/components/GenerateGifForm.js
Comment (phillipson808) [reply src/components/GenerateGifForm.js]: Done

File: src/components/GenerateGifForm.js
Comment (phillipson808) [reply src/components/GenerateGifForm.js]: Done

File: src/components/GenerateGifForm.js
Comment (phillipson808) [reply src/components/GenerateGifForm.js]: Done

File: src/constants/action-types.js
Comment (phillipson808) [reply src/constants/action-types.js:16]: Done

File: src/components/GenerateGifForm.js
Comment (phillipson808) [reply src/components/GenerateGifForm.js]: Done

File: src/components/GenerateGifForm.js
Comment (phillipson808) [reply src/components/GenerateGifForm.js]: Done

File: src/components/Preview.js
Comment (phillipson808) [reply src/components/Preview.js:93]: Done

File: src/reducers/images.js
Comment (phillipson808) [reply src/reducers/images.js]: Done

File: src/components/GenerateGifForm.js
Comment (misscoded) [reply src/components/GenerateGifForm.js]: Better to use a switch statement.

Comment (lopezh18) [reply src/components/GenerateGifForm.js]: Done

**Comment by misscoded** [comment]
@ctlusto I believe this is ready for your review.

File: src/reducers/images.js
Comment (ctlusto) [reply src/reducers/images.js]: Uh...no. That's probably leftover from something I was doing earlier.

File: src/components/Preview.js
Comment (ctlusto) [inline src/components/Preview.js]: I'm not a huge fan of these comments. I'm not entirely sure what they're for, and it's possible they're stale already? This is the only place that `defaultColor` shows up in the codebase. Let's just get rid of these.

File: src/components/Preview.js
Comment (ctlusto) [reply src/components/Preview.js:93]: Yeah, this is a good call.

File: src/reducers/images.js
Comment (ctlusto) [inline src/reducers/images.js]: I assume this was a typo for `#000`. Looks like the color picker always generates six-digit hex colors, so let's use `#000000` here as well.

**Review by ctlusto — CHANGES_REQUESTED** [review]
Overall this is really great. Couple of small things I'd like cleaned up, but it will take two minutes.

Nice job. 🙇

File: src/components/Preview.js
Comment (lopezh18) [reply src/components/Preview.js]: Done

**Pull request merged**
